libdbus-sys/build_vendored.rs

```diff
@@ -0,0 +1,389 @@
+
+// Location of dbus-daemon socket
+const DBUS_SYSTEM_BUS_DEFAULT_ADDRESS: &str = "/run/dbus/system_bus_socket";
+const DBUS_SESSION_SOCKET_DIR: &str = "/tmp";
+const DBUS_MACHINE_UUID_FILE: &str = "/etc/machine-id";
+
+/// Source Files Required
+/// Reference: https://github.com/freedesktop/dbus/blob/master/dbus/meson.build#L31
+const SOURCE_FILES: &'static [&'static str] = &[
+    "./vendor/dbus/dbus/dbus-address.c",
+    "./vendor/dbus/dbus/dbus-auth.c",
+    "./vendor/dbus/dbus/dbus-bus.c",
+    "./vendor/dbus/dbus/dbus-connection.c",
+    "./vendor/dbus/dbus/dbus-credentials.c",
+    "./vendor/dbus/dbus/dbus-errors.c",
+    "./vendor/dbus/dbus/dbus-keyring.c",
+    "./vendor/dbus/dbus/dbus-marshal-byteswap.c",
+    "./vendor/dbus/dbus/dbus-marshal-header.c",
+    "./vendor/dbus/dbus/dbus-marshal-recursive.c",
+    "./vendor/dbus/dbus/dbus-marshal-validate.c",
+    "./vendor/dbus/dbus/dbus-message.c",
+    "./vendor/dbus/dbus/dbus-misc.c",
+    "./vendor/dbus/dbus/dbus-nonce.c",
+    "./vendor/dbus/dbus/dbus-object-tree.c",
+    "./vendor/dbus/dbus/dbus-pending-call.c",
+    "./vendor/dbus/dbus/dbus-resources.c",
+    "./vendor/dbus/dbus/dbus-server-debug-pipe.c",
+    "./vendor/dbus/dbus/dbus-server-socket.c",
+    "./vendor/dbus/dbus/dbus-server.c",
+    "./vendor/dbus/dbus/dbus-sha.c",
+    "./vendor/dbus/dbus/dbus-signature.c",
+    "./vendor/dbus/dbus/dbus-syntax.c",
+    "./vendor/dbus/dbus/dbus-threads.c",
+    "./vendor/dbus/dbus/dbus-timeout.c",
+    "./vendor/dbus/dbus/dbus-transport-socket.c",
+    "./vendor/dbus/dbus/dbus-transport.c",
+    "./vendor/dbus/dbus/dbus-watch.c",
+];
+
+/// Internal files
+/// Reference: https://github.com/freedesktop/dbus/blob/master/dbus/meson.build#L67
+const INTERNAL_FILES: &'static [&'static str] = &[
+    "./vendor/dbus/dbus/dbus-dataslot.c",
+    "./vendor/dbus/dbus/dbus-file.c",
+    "./vendor/dbus/dbus/dbus-hash.c",
+    "./vendor/dbus/dbus/dbus-internals.c",
+    "./vendor/dbus/dbus/dbus-list.c",
+    "./vendor/dbus/dbus/dbus-marshal-basic.c",
+    "./vendor/dbus/dbus/dbus-memory.c",
+    "./vendor/dbus/dbus/dbus-mempool.c",
+    "./vendor/dbus/dbus/dbus-pipe.c",
+    "./vendor/dbus/dbus/dbus-string.c",
+    "./vendor/dbus/dbus/dbus-sysdeps.c",
+];
+
+/// Unix platform files
+/// References:
+/// - https://github.com/freedesktop/dbus/blob/master/dbus/meson.build#L139
+/// - https://github.com/freedesktop/dbus/blob/master/dbus/meson.build#L144
+const UNIX_SOURCES: &'static [&'static str] = &[
+    // Lib
+    "./vendor/dbus/dbus/dbus-uuidgen.c",
+    "./vendor/dbus/dbus/dbus-server-unix.c",
+
+    // Shared
+    "./vendor/dbus/dbus/dbus-file-unix.c",
+    "./vendor/dbus/dbus/dbus-pipe-unix.c",
+    "./vendor/dbus/dbus/dbus-sysdeps-pthread.c",
+    "./vendor/dbus/dbus/dbus-sysdeps-unix.c",
+    "./vendor/dbus/dbus/dbus-transport-unix.c",
+    "./vendor/dbus/dbus/dbus-userdb.c",
+];
+
+/// Taken directly from the libdbus meson config
+const CWARNINGS: &'static [&'static str] = &[
```

Comment:
Should also have reference

Reply:
Documented in 84b3ddf35a391205ce7340325c5b01879ced1359